### ISSUE TITLE
False positive W0613 with keyword only argument

### ISSUE THREAD

**Issue by DevinShanahan** [PyCQA/pylint#5771]
### Bug description

Using a keyword-only argument is triggering a W0613(unused-argument) warning where it is not occurring if the (unused) argument is a keyword argument. Below, `B.add` does not issue the warning, but `D.add` does.

kwoa.py
```python
# pylint: disable=C*,R*
class A:
    def add(self, a, b=0):
        return a + b

class B(A):
    def add(self, a, b=0):
        return None

class C:
    def add(self, a, *, b=0):
        return a + b

class D(C):
    def add(self, a, *, b=0):
        return None
```

### Command used

```shell
pylint kwoa.py
```

### Pylint output

```shell
************* Module kwoa
kwoa.py:16:24: W0613: Unused argument 'b' (unused-argument)
```

### Expected behavior

I believe there are two possibilities here but I had been operating under the assumption that this warning would not occur for `D.add` given that is not the behavior for `B.add`. The other option is that a warning should also be issued `B.add`.

### Pylint version

```shell
pylint 2.12.2
astroid 2.9.0
Python 3.9.5 (default, May 18 2021, 12:31:01) 
[Clang 10.0.0 ]
```

**Comment by jacobtylerwalls** [comment]
Thanks for the report. Based on this [comment](https://github.com/PyCQA/pylint/blob/d7013f6799d28a69dbbf3329da7867a8afacce82/pylint/checkers/variables.py#L2342) I think you are correct that this is not supposed to warn. I have a diff for `astroid` that fixes it and hope to PR it soon.

**Comment by jacobtylerwalls** [comment]
PR adding the regression test should also simplify this to just `argnames()`: https://github.com/PyCQA/pylint/commit/01cc02d5e
